src/book/04-general/01-git.mdx

```diff
@@ -0,0 +1,59 @@
+
+Suppose you have a file you have made which you would like to add to a NUbots repository, called `my_file.cpp`, as part of your project to make the robot stand up.
+
+In the terminal, change directory to where you have the repo cloned.
+
+### Select Branch
+
+If you are working within an existing project, then you need to find out which branch the work is on. To list the branches, use `git branch`. When you have found the branch you want to push to, swap to it:
+
+```sh
+git checkout <branch name>
+```
+
+If you are not working on an existing project, you need to make a branch to commit your changes to. The format for branch names is `surname/project_name`. So for this example it could be `citizen/stand_up`.
+
+### Making a Pull Request
+
+Once you have a set of commits on your branch which you think should be added to the main codebase, you can make a Pull Request (PR). PRs are where users ask the repository to pull their branch into the `master` branch.
+
+Github has documentation about PRs [here.](#https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
+
+Title your PR with a description of the changes that it will make. For this example we could use `Add stand-up script`. If you need to give more detail about what the code does, put that as a comment.
+
+To be approved, you should add reviewers to the pull request. If you have a project mentor, they should be a reviewer, as well as someone else who knows about that part of the codebase. Once the reviewers have approved the changes, the changes can be merged into the master branch.
+
+### Code Review
+
+The reviewers will give constructive feedback about your code, by making comments on code snippets from your changes. You can respond to these on Github. They can also directly suggest changes to specific sections. Additional commits you make to the branch will update the PR and will reset their review of changed sections.
+
+Once the reviewers are happy with it, and all of the comments are resolved, you can `Squash and Merge`. Squashing removes detailed commit descriptions, so that `master` is not messy with every commit ever. Once the code has been squashed and merged, it is in the codebase and you have succeeded in contributing code to NUbots.
+
+## Commit Etiquette
+
+Commits should make minor changes, and the changes in each commit should have a common theme. A general mantra for contributing code is `commit often, push once`, but this isn't strict at all. Committing often means that the code is built up incrementally. Pushing once means that you don't spam all the other contributors with notifications.
+
+## Formatting and buildkite
+
+Before making a pull request, it is important that your code adheres to the formatting rules. For branches in the NUbots repo, running `./b format` before your last commit should fix formatting issues. Code in the NUbook repo must be formatted with `yarn format` before it can be committed.
```

Comment:
Is there a similar thing for NUsight2 @JosephusPaye?

Reply:
For NUsight, it's `yarn eslint` to find problems and `yarn eslint:fix` to fix.